server/map_rpc_server.go

```diff
@@ -309,9 +309,18 @@ func (t *TrillianMapServer) getLeavesByRevision(ctx context.Context, mapID int64
 func (t *TrillianMapServer) SetLeaves(ctx context.Context, req *trillian.SetMapLeavesRequest) (*trillian.SetMapLeavesResponse, error) {
 	ctx, spanEnd := spanFor(ctx, "SetLeaves")
 	defer spanEnd()
-
 	t.setLeafCounter.Add(float64(len(req.Leaves)), strconv.FormatInt(req.MapId, 10))
 
+	// TODO(pavelkalinnikov): Factor out a CreateRevision RPC, and merge it with
+	// InitMap which creates the 0th revision.
+	if len(req.Leaves) == 0 {
+		newSMR, err := t.addRevision(ctx, req.MapId, req.Revision, req.Metadata)
```

Comment:
If `Revision == 0` then this will have the same effect as `InitMap`. We should probably disallow it for now?